src/core.jsx

```diff
@@ -31,20 +31,20 @@ export default config => WrappedComponent => class extends Component {
   }
 
   format = (nextProps) => {
-    if (!this.context.store
-      || !this.context.store.getState()
-      || !this.context.store.getState().locale) {
-      throw new Error('/ HOC k-intl / locale is not readable. make sure that this one is available at the root of your redux store')
-    }
+    /* take locale on `config.locale` reducer */
+    const locale = getLocale(this.context)
 
     /* not change labels when the sub store local not change */
-    if (this.context.store.getState().locale === this.locale) return
-    this.locale = this.context.store.getState().locale
+    if (locale === this.locale) return
```

Comment:
this is not the locale you have to cache but the format.
And you SHOULD call setState when props changes since it changes the labels